From 9d9a725aa8c51d16d5638e9228b721b547f7f455 Mon Sep 17 00:00:00 2001
From: Mathieu Bivert <mathieu.bivert@intel.com>
Date: Wed, 14 Aug 2013 15:36:16 +0200
Subject: [PORT FROM MAIN]wifi/crashtool: log reason_code when deasossiating from an access point

BZ: 130858

Trace reason_code all the time, unless the user choose to leave the AP.

Change-Id: I5a93e285ff73045acaea3a3e5ac76a965708bd8b
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Mathieu Bivert <mathieu.bivert@intel.com>
Reviewed-on: http://android.intel.com:8080/125559
Reviewed-by: Trivelly, Jean <jean.trivelly@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Tested-by: Lestienne, ChristopheX <christophex.lestienne@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 wpa_supplicant/wpa_supplicant.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/wpa_supplicant/wpa_supplicant.c b/wpa_supplicant/wpa_supplicant.c
index eea4e8c..e2df913 100644
--- a/wpa_supplicant/wpa_supplicant.c
+++ b/wpa_supplicant/wpa_supplicant.c
@@ -1815,6 +1815,12 @@ void wpa_supplicant_deauthenticate(struct wpa_supplicant *wpa_s,
 		wpa_s->current_bss->is_hidden = 1;
 	}
 
+	char b[16];
+
+	if (reason_code != WLAN_REASON_DEAUTH_LEAVING) {
+		snprintf(b,sizeof(b), "%d", reason_code);
+		lct_log(CT_EV_INFO, "cws.wifi", "deauthenticate", 0, b);
+	}
 
 	wpa_dbg(wpa_s, MSG_DEBUG, "Request to deauthenticate - bssid=" MACSTR
 		" pending_bssid=" MACSTR " reason=%d state=%s",
-- 
1.7.4.1

